beancount-black fails to build

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build .#beancount-black`

### Build log
```
=========================== short test summary info ============================
FAILED tests/test_formatter.py::test_format_date_directive[column_widths5-1970-01-01  balance     Assets:Bank  12.3>
FAILED tests/test_main.py::test_main[simple.bean-simple.bean] - AttributeError: 'Tree' object has no attribute 'val>
FAILED tests/test_main.py::test_main[cost_and_price.bean-cost_and_price.bean] - lark.exceptions.UnexpectedToken: Un>
FAILED tests/test_main.py::test_main[sections.bean-sections.bean] - lark.exceptions.UnexpectedToken: Unexpected tok>
FAILED tests/test_main.py::test_main[column_width.bean-column_width.bean] - AttributeError: 'Tree' object has no at>
FAILED tests/test_main.py::test_main[txn.bean-txn.bean] - AttributeError: 'Tree' object has no attribute 'value'
========================= 6 failed, 29 passed in 2.28s =========================
```

Full build log: https://gist.github.com/kilianar/14483781791eb85039adb81bb24d055a

### Additional context
This error occurs because the version of [beancount-parser](https://github.com/LaunchPlatform/beancount-parser) in nixpkgs is 0.1.23 and `beancount-black` requires a prior version. Setting `doCheck = false;` resolves the build error but leads to a runtime error that crashes the program. There is an [upstream issue](https://github.com/LaunchPlatform/beancount-black/issues/8) but they seem to be rather unresponsive. 

Everything does work as intended when the version of `beancount-parser` is manually pinned to 0.1.21.


### Notify maintainers
@ambroisie

--- a/pkgs/development/python-modules/beancount-black/default.nix
+++ b/pkgs/development/python-modules/beancount-black/default.nix
@@ -10,7 +10,7 @@
 
 buildPythonPackage rec {
   pname = "beancount-black";
-  version = "0.1.13";
+  version = "0.1.14";
 
   disabled = pythonOlder "3.9";
   format = "pyproject";
@@ -19,7 +19,7 @@ buildPythonPackage rec {
     owner = "LaunchPlatform";
     repo = "beancount-black";
     rev = version;
-    sha256 = "sha256-jhcPR+5+e8d9cbcXC//xuBwmZ14xtXNlYtmH5yNSU0E=";
+    hash = "sha256-4ooMskwPJJLJBfPikaHJ4xuwR1x478ecYWZdIE0UAK8=";
   };
 
   buildInputs = [

